client/cmd/daemon/filesystem.go

```diff
@@ -19,3 +24,122 @@ func execPath() (string, error) {
 	return exe, nil
 }
 
+func envFileAlreadyExist(path string) bool {
+	if _, err := os.Stat(path); err == nil {
+		return true
+	}
+	return false
+}
+
+func envFileExist() (string, error) {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return "", err
+	}
+	envPath := filepath.Join(home, ".config", "hoop.conf")
+	if envFileAlreadyExist(envPath) {
+		return envPath, nil
+	}
+	return "", fmt.Errorf("env file not found")
+}
+
+func LoadEnvFile(path string) (map[string]string, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	env := make(map[string]string)
+	scanner := bufio.NewScanner(f)
+	home, _ := os.UserHomeDir()
+
+	for scanner.Scan() {
+		line := strings.TrimSpace(scanner.Text())
+		if line == "" || strings.HasPrefix(line, "#") {
+			continue
+		}
+		parts := strings.SplitN(line, "=", 2)
+		if len(parts) != 2 {
+			continue // skip malformed line
+		}
+		key := strings.TrimSpace(parts[0])
+		val := strings.TrimSpace(parts[1])
+
+		if strings.HasPrefix(val, "~/") {
+			val = filepath.Join(home, val[2:])
+		}
+
+		env[key] = val
+	}
+	if err := scanner.Err(); err != nil {
+		return nil, err
+	}
+	return env, nil
+}
+
+func createEnvFileIfNotExists(env map[string]string) (string, error) {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		log.Errorf("error getting user home dir: %v", err)
+		return "", err
+	}
+	envDir := filepath.Join(home, ".config")
+	envPath := filepath.Join(envDir, "hoop.conf")
+
+	if err := os.MkdirAll(envDir, 0o755); err != nil {
+		log.Errorf("error creating config dir: %v", err)
+		return "", err
+	}
+
+	content := ""
+	for k, v := range env {
+		content += fmt.Sprintf("%s=%s\n", k, v)
+	}
+
+	if err := writeFileIfNotExists(envPath, content, 0o600); err != nil {
+		log.Errorf("error writing env file: %v", err)
+		return "", err
+	}
+	return envPath, nil
+}
+
+func writeFileIfNotExists(path, content string, perm os.FileMode) error {
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		if err := os.WriteFile(path, []byte(content), perm); err != nil {
+			log.Errorf("Error writing the file %s: %v", path, err)
+			return err
+		}
+	} else if err != nil {
+		log.Errorf("Error stat file %s: %v", path, err)
+		return err
+	}
+	return nil
```

Comment:
```suggestion
	_, err := os.Stat(path)
	if err == nil {
	    // file already exists
		return nil
	}
	if !os.IsNotExist(err) {
		log.Errorf("Error stating file %s: %v", path, err)
		return err
	}

	if err := os.WriteFile(path, []byte(content), perm); err != nil {
		log.Errorf("Error writing file %s: %v", path, err)
		return err
	}
	return nil
```
Maybe a bit more idiomatic